index.js

```diff
@@ -118,6 +118,24 @@ function doTransform(checker, targetVersion, k) {
           )
         );
       }
+    } else if (
+      semver.lt(targetVersion, "3.6.0") &&
+      ((ts.isTypeReferenceNode(n) && ts.isIdentifier(n.typeName) && n.typeName.escapedText === "IteratorResult") ||
+        (ts.isExpressionWithTypeArguments(n) &&
+          ts.isIdentifier(n.expression) &&
+          n.expression.escapedText === "IteratorResult"))
+    ) {
+      const symbol = checker.getSymbolAtLocation(ts.isTypeReferenceNode(n) ? n.typeName : n.expression);
+      const typeArguments = n.typeArguments;
+      if (
+        semver.lt(targetVersion, "3.6.0") &&
+        symbol &&
+        symbol.declarations.length &&
+        symbol.declarations[0].getSourceFile().fileName.includes("node_modules/typescript/lib/lib") &&
```

Comment:
This is complicated enough to extract into a function now that it's happened twice, I think.

Reply:
Good point, I factored it into a `isStdLibSymbol` function.